testing/scripts/test_api.sh

```
@@ -1027,6 +1027,86 @@ else
     skip "Scan API tests — no test subnet available (SUBNET_ID not set)"
 fi
 
+# ====================================================================
+log "=== Devices API ==="
+# ====================================================================
+
+# POST device
+call_api POST "devices" '{"name":"api-test-router","type":"router","vendor":"Cisco","model":"ISR4321","serial":"TST001","note":""}'
+assert_http 201 "POST device"
+DEV_ID=$(python3 -c "import sys,json; d=json.load(sys.stdin); print(d.get('device',{}).get('id',''))" <<< "$BODY" 2>/dev/null || echo "")
+[[ -n "$DEV_ID" && "$DEV_ID" != "None" ]] && pass "POST device: id=$DEV_ID" || fail "POST device: no id in response"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)